### PR TITLE
[KEYCLOAK-11771] Add missing type definitions for config

### PR DESCRIPTION
**Pull request by jonkoops**
This PR adds the missing TypeScript definitions needed for configuration options provided to the constructor.

It does not include type definitions for the `oidcProvider` option introduced in https://github.com/keycloak/keycloak/pull/4978 which is currently not documented.

Depends on: https://github.com/keycloak/keycloak/pull/6454

### CODE REVIEW

**Comment by jonkoops** [comment]
@stianst Would it be possible to get this in before version 8 is released? We'd like to move the missing type definitions from keycloak-angular to keycloak-js if possible.

File: adapters/oidc/js/src/main/resources/keycloak.d.ts
Comment (abstractj) [inline adapters/oidc/js/src/main/resources/keycloak.d.ts]: @jonkoops overall looks good to me. I just wonder if we really need a separate interface just to store secrets, instead of just use `credentials?: string;`

@ssilvert what do you think?

Comment (jonkoops) [reply adapters/oidc/js/src/main/resources/keycloak.d.ts]: Could be an inline thing, in this case I am just adding it to ensure that you could also use this interface separately as a user.

Comment (mhajas) [reply adapters/oidc/js/src/main/resources/keycloak.d.ts]: In documentation we have this line: 
````
To use the JavaScript adapter you must first create a client for your application in the Keycloak Administration Console. Make sure public is selected for Access Type.
````

Clients which are used with Javascript adapter should be public which means they don't have credentials. So should this be removed or I missed something?

Comment (jonkoops) [reply adapters/oidc/js/src/main/resources/keycloak.d.ts]: @mhajas If that is indeed the case, should this entire functionality not be removed from the JavaScript Adapter in the first place?

Comment (ssilvert) [reply adapters/oidc/js/src/main/resources/keycloak.d.ts]: Since KeycloakCredentials just wraps a string I don't see how it adds any value.

The rest looks good.

**Review by abstractj — CHANGES_REQUESTED** [review]
@jonkoops based on the latest comments from @ssilvert could you please update your PR so we can get it merged?

File: adapters/oidc/js/src/main/resources/keycloak.d.ts
Comment (mhajas) [reply adapters/oidc/js/src/main/resources/keycloak.d.ts]: @jonkoops this discussion is probably out of scope for this PR, but it seems useless to me. I will start a thread in dev ML.

**Comment by jonkoops** [comment]
@abstractj  Done. Since there is some confusion about whether it is even a good idea to support the credentials feature I've removed the typings for it entirely.

@mhajas Can you start a discussion on keycloak-dev about this? If the feature is not secure we should remove this functionality entirely. If need be I can create an issue for this on the tracker.

**Comment by mhajas** [comment]
@jonkoops Done.

In my opinion, credentials should stay there for now, as we are using it in the keycloak.js file. If we will decide in the discussion, to remove it, we can remove it then.

**Comment by jonkoops** [comment]
Agreed, let's have a discussion and see where we can go with this.

**Comment by stianst** [comment]
I'd just remove the credentials as it's not documented and shouldn't be used. We can discuss separately if/when we remove from keycloak.js

**Comment by jonkoops** [comment]
@abstractj Now that the `credentials` option has been removed, should we go ahead and merge this one as well?

**Review by abstractj — APPROVED** [review]
@mhajas @stianst looks good to me. But I will wait for your approval before merge it.

**Comment by abstractj** [comment]
@jonkoops thanks for these changes.

**Review by mhajas — APPROVED** [review]
@jonkoops Thank you.

**Comment by jonkoops** [comment]
Thanks guys, appreciate it!

**Pull request merged**
